File: backend/src/main/java/com/example/backend/domain/group/exception/GroupException.java
package com.example.backend.domain.group.exception;

import org.springframework.http.HttpStatus;

public class GroupException extends RuntimeException{
    private final GroupErrorCode groupErrorCode;

    public GroupException(GroupErrorCode groupErrorCode) {
        super(groupErrorCode.getMessage());
        this.groupErrorCode = groupErrorCode;
    }

    public HttpStatus getStatus() {
        return groupErrorCode.getHttpStatus();
    }

    public String getCode() {
        return groupErrorCode.getCode();
    }
}

File: backend/src/main/java/com/example/backend/domain/group/exception/GroupException.kt
package com.example.backend.domain.group.exception

import org.springframework.http.HttpStatus

class GroupException(private val groupErrorCode: GroupErrorCode) : RuntimeException(groupErrorCode.message) {
    val status: HttpStatus
        get() = groupErrorCode.httpStatus

    val code: String
        get() = groupErrorCode.code
}
